cmd/outline-ss-server/main.go

```diff
@@ -130,48 +134,101 @@ func (s *SSServer) loadConfig(filename string) error {
 		return fmt.Errorf("failed to load config (%v): %w", filename, err)
 	}
 
```

Comment:
This logic was already quite complicated. I need more time to review this part.

I also wonder if there's a better approach. In Caddy, it just instantiates the new objects, which we could do here as well, then shuts downs the old stuff after the new stuff is serving. Then you don't need to do config diffs.

The main issue is the port conflicts. The new listener will fail if there's an old listener on the same address.  I think Caddy addresses this by using SO_REUSEPORT/SO_REUSEADDR, but that is a bit [tricky](https://stackoverflow.com/questions/14388706/how-do-so-reuseaddr-and-so-reuseport-differ) to get right across platforms.